salt/state.py

```diff
@@ -2783,10 +2783,31 @@ def call_listen(self, chunks, running):
         running.update(errors)
         return running
 
+    def inject_default_call(self, high):
+        '''
+        Sets .call function to a state, if not there.
+
+        :param high:
+        :return:
+        '''
+        for chunk in high:
+            state = high[chunk]
+            for state_ref in state:
+                needs_default = True
+                for argset in state[state_ref]:
+                    if isinstance(argset, six.string_types):
+                        needs_default = False
+                        break
+                if needs_default:
+                    order = state[state_ref].pop(-1)
+                    state[state_ref].append('__call__')
+                    state[state_ref].append(order)
+
     def call_high(self, high, orchestration_jid=None):
         '''
         Process a high data call and ensure the defined states.
         '''
+        self.inject_default_call(high)
```

Comment:
Do you have evidence we aren't careful? :wink: It actually doesn't clash at all. Performance wise it hardly adds even a nanofraction of a second per minion. All what it does, just looks if there is function defined on top level (and usually it is). If you did not defined one, it adds `__call__` reference and quits. Or just quits (default). BTW, do we have already some numbers that proves it affects performance?

But on the other hand this it allows to get rid of ID-per-function SLS style that forces to write repetitive states, yet perfectly preserving original way without clashes.

Reply:
@cachedout some update: I've measured performance. It is actually the same, except new version (single-id) in some very small fraction a bit faster in average. But I don't know how true this is, since many factors are involved.